src/main/java/de/tum/in/www1/artemis/service/CourseService.java

```diff
@@ -333,7 +355,7 @@ public void cleanupCourse(Long courseId) {
         }
 
         // Clean up exams
-        var exams = courseRepository.findByIdWithLecturesAndExamsElseThrow(course.getId()).getExams();
+        var exams = courseRepository.findByIdWithLecturesAndLectureUnitsAndExamsElseThrow(course.getId()).getExams();
```

Comment:
Also seems to be the wrong call. We should find the exams by course id

Reply:
Done in b46299e24c6f423f6b6aef00b7c4c2077c526241